### ISSUE TITLE
FAHControl: Log segments repeated when the warnings and errors filter is active

### ISSUE THREAD

**Issue by sombraguerrero** [FoldingAtHome/fah-issues#1230]
`*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
03:27:05:WARNING:WU02:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
03:27:05:WARNING:WU02:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
03:27:05:WARNING:WU02:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
03:27:05:WARNING:WU02:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
03:27:05:WARNING:WU02:FS00:AS lowered CPUs from 7 to 6
*********************** Log Started 2018-05-21T00:12:36Z ***********************
00:13:07:WARNING:WU00:FS00:Changed SMP threads from 6 to 7 this can cause some work units to fail
00:13:07:WARNING:WU00:FS00:AS lowered CPUs from 7 to 6
03:27:05:WARNING:WU02:FS00:AS lowered CPUs from 7 to 6
`

**Comment by bollix47** [comment]
I see this in the log tab of FAHControl when selecting Warnings & Errors.  The first time after restarting is okay but subsequent attempts to check that selection will result in two or more repeats of the same messages with the same times.

**Comment by jcoffland** [comment]
Sounds like a bug in FAHControl.  Does this also occur in earlier versions?

**Comment by bollix47** [comment]
Yes, it was the same in 7.4.4.

**Comment by bb30994** [comment]
Is this a problem associate with the configuration mentioned in your other bug report? (files are on D:)
(@jcoffland: Maybe it can't be reproduced if the default file locations are used.)

**Comment by bollix47** [comment]
All of my setups are using the default locations and I can reproduce the problem at will.

**Comment by bollix47** [comment]
It is unclear to me whether this only happens when monitoring other setups or if it happens on the same setup.  ie I know it happens when monitoring other folding setups from one main FAHControl but I'm not sure it happens when running FAHControl directly on that folding setup. Just now I noticed a duplication of data under Warnings & Errors on the FAHControl that is monitoring my systems but when I went to the actual folding setup and ran FAHControl the duplication was not shown.  Perhaps a communication buffer is not being cleared?

**Comment by sombraguerrero** [comment]
@jcoffland No, this issue does not have anything to do with the other issue (see my most recent comment). The duplication on my system is local, as in the local FAHControl on that same box. My two folding computers both monitor each other on FAHControl. I have not yet checked the FAHControl on my other system to see if the duplication from my main system appears there. I will try to do that once they both are running long  enough to generate a fair amount of activity for the day.

**Comment by sombraguerrero** [comment]
Okay, I can now corroborate @bollix47. When I checked the log in FAHControl for my main box via my second system, the duplication was not there. Additionally, when I went back to my main box and checked it again locally, the duplication was not there. So I definitely agree that it's some sort of caching/memory buffer sort of thing the first time the logs are punched up locally on FAHControl.
